### PR TITLE
[TraceDecoder] Fix trace decoder failing

### PR DESCRIPTION
**Pull request by pankore**
* Fixes issue #23144
* mJsonBuffer not cleared before calling PrepareMessage, causing above issue to happen
* Added ResetTraceStream to clear buffer

### CODE REVIEW

**Comment by bzbarsky-apple** [comment]
@tcarmelveilleux @vivien-apple please take a look?

File: examples/common/tracing/TraceDecoder.h [hunk 1]
```diff
@@ -44,6 +44,13 @@ class TraceDecoder : public TraceStream
         snprintf(buffer, sizeof(buffer), "    %s\t %s", tag.c_str(), data.c_str());
         ReadString(buffer);
     }
+    void CleanJsonBuf(void) override
```
Comment (andy31415) [inline examples/common/tracing/TraceDecoder.h:47]: add whitespace above.

Do not use `(void)` just use `()`

File: examples/common/tracing/TraceHandlers.h [hunk 2]
```diff
@@ -53,6 +54,8 @@ class TraceStreamLog : public TraceStream
 
     void FinishEvent() override { ChipLogAutomation("TraceStream:%s", mStreamLine.c_str()); }
 
+    void CleanJsonBuf(void) override {}
```
Comment (andy31415) [inline examples/common/tracing/TraceHandlers.h:57]: use `()` instead of `(void)` throughout.

**Review by andy31415 — APPROVED** [review]

File: examples/common/tracing/TraceDecoder.h (hunk 1 above)
Comment (pankore) [reply examples/common/tracing/TraceDecoder.h]: Fixed

File: examples/common/tracing/TraceHandlers.h (hunk 2 above)
Comment (pankore) [reply examples/common/tracing/TraceHandlers.h]: Fixed

**Review by woody-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@pankore Can you please fix the CI problems?

**Comment by woody-apple** [comment]
Closing old stale PRs, please feel free to re-open if you're still working on this!

**Pull request closed without merging**